pyomo/contrib/pynumero/algorithms/solvers/cyipopt_solver.py

```diff
@@ -148,7 +148,7 @@ def _cyipopt_importer():
     'Internal_Error': TerminationCondition.internalSolverError,
 }
 
-class CyIpoptProblemInterface(object, metaclass=abc.ABCMeta):
+class CyIpoptProblemInterface(cyipopt.Problem, metaclass=abc.ABCMeta):
```

Comment:
This fails if cyipopt is not available. We could get around this with
```
cyipopt_Problem = cyipopt.Problem if cyipopt_available else object
```
at the top of the file, but then we get a confusing error message when we try to call `CyIpoptProblemInterface.solve`. Maybe do this but raise an error in `__init_subclass__` if we're not an instance of `cyipopt.Problem`?

Reply:
What about using `cyipopt_Problem` as you suggest, then raising a sensible exception in `__init__` if `not cyipopt_available`?

Reply:
Not bad, we just have to remember to call the right combination of `super().__init__`s in subclasses. But I could move the `cyipopt.Problem.__init__` call into `CyIpoptProblemInterface.__init__`, so subclass developers only have to remember to call one `super().__init__`.